src/sentry/integrations/discord/client.py

```diff
@@ -57,6 +64,50 @@ def get_guild_name(self, guild_id: str) -> str:
         response = self.get(GUILD_URL.format(guild_id=guild_id), headers=self.prepare_auth_header())
         return response["name"]  # type: ignore
 
+    def get_access_token(self, code: str, url: str):
+        data = {
+            "client_id": self.application_id,
+            "client_secret": self.client_secret,
+            "grant_type": "authorization_code",
+            "code": code,
+            "redirect_uri": url,
+            "scope": "identify",
+        }
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        try:
```

Comment:
Typically the clients are purely for calling endpoints, and the code who calls these methods deal with the error handling and logging. I'd remove the try/catch here and leave it in `integration.py`.